src/engine/strat_engine/util.rs

```diff
@@ -72,3 +79,32 @@ pub fn set_uuid(devnode: &Path, uuid: Uuid) -> EngineResult<()> {
         Err(EngineError::Engine(ErrorEnum::Error, err_msg))
     }
 }
+
+/// Lookup the WWN from the udev db using the device node eg. /dev/sda
+pub fn hw_lookup(dev_node_search: &Path) -> strat_engine_util::Result<Option<String>> {
```

Comment:
Given that we have no idea what error system we'll be using, please don't use error_chain.

Try instead adding a new error constructor for Udev errors to EngineError. Implement From for this constructor and it should be auto-promoted by the ```?``` operator. Transform ```chain_err(...)?``` to ```?``` everywhere and it may all compile.